websites/G/Google Play/presence.ts

```diff
@@ -0,0 +1,133 @@
+    ENTERTAINMENT: "Browsing for entertaiment apps",
+    SOCIAL: "Browsing for social apps",
+    SPORTS: "Browsing for sport apps",
+    EVENTS: "Browsing for eventing apps",
+    FINANCE: "Browsing for financing apps",
+    PHOTOGRAPHY: "Browsing for photography apps",
+    FOOD_AND_DRINK: "Browsing for food apps",
+    WATCH_FACE: "Browisng for android watch faces"
+  };
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "logo"
+    },
+    query = new URLSearchParams(document.location.search),
+    paths: string[] = document.location.pathname.split("/");
+  if (!paths[0]) paths.shift();
+
+  if (paths[0] === "settings") presenceData.details = "Viewing settings";
+  if (paths[0] === "apps" && !paths[1])
+    presenceData.details = "Viewing my apps";
+  else if (paths[0] === "wishlist") presenceData.details = "Viewing wishlist";
+  else if (paths[0] === "store") {
+    if (paths[1] === "family" && paths[2] === "create")
+      presenceData.details = "Creating family";
+    if (paths[1] === "paymentmethods")
+      presenceData.details = "Viewing payment methods";
+    else if (paths[1] === "myplayactivity")
+      presenceData.details = "Viewing my activity";
+    else if (paths[1] === "search") {
+      const q = query.get("q");
+      presenceData.details = `Searching${q ? ` for ${q}` : ""}`;
+    } else if (paths[1] === "account") {
+      if (!paths[2] || paths[2] === "rewards")
+        presenceData.details = "Viewing rewards";
+      else if (paths[2] === "orderhistory")
+        presenceData.details = "Viewing order hystory";
+      else if (paths[2] === "family") presenceData.details = "Viewing family";
+      else if (paths[2] === "subscriptions")
+        presenceData.details = "Viewing subscriptions";
+    } else if (paths[1] === "apps") {
+      if (!paths[2]) presenceData.details = "Browsing for apps";
+      else if (paths[2] === "top")
+        presenceData.details = "Browsing for the TOP apps";
+      else if (paths[2] === "new")
+        presenceData.details = "Browsing for the new apps";
+      else if (paths[2] === "details") {
+        if (query.get("id")) {
+          const body = document.querySelector<HTMLDivElement>(
+              "body > div#fcxH9b > div.WpDbMd > c-wiz > div.T4LgNb > div.ZfcPIb > div.UTg3hd > div.JNury > main.LXrl4c > c-wiz > c-wiz > div.oQ6oV"
+            ),
+            logo = body?.querySelector<HTMLImageElement>(
+              "div.hkhL9e > div.xSyT2c > img.T75of"
+            )?.src,
+            name = body?.querySelector<HTMLSpanElement>(
+              "div.D0ZKYe > div.rlnrKc > div.sIskre > c-wiz > h1.AHFaub > span"
+            ).textContent;
+
+          if (logo) presenceData.largeImageKey = await getShortURL(logo);
+
+          presenceData.details = `Viewing ${name ? name : "app"}`;
+          presenceData.buttons = [
+            {
+              label: "View app",
+              url: await getShortURL(document.location.href)
```

Comment:
The max length here is 512 not 256. I doubt it'd get that long?

Reply:
I really don't know if the url length of the image will be more than 512, but the `getShorURL` method returns the original url if it's not more than 512, I know that will make the presence slower.
Should I remove it?

Reply:
I meant buttons' url not imageKey

Reply:
The default app url should be `https://play.google.com/store/apps/details?id=com.instagram.android` for example

But if Google adds some queries, and sometimes they put a lot of queries, the URL will get more than 512 characters (ig)